src/store/createStore.ts

```diff
@@ -127,13 +135,16 @@ export const createStore = (options: CreateStoreOptions) =>
                 ...value,
               };
             }),
-          updatePublicKey: (value: string) =>
+          updateKeypair: (publicKey: string, secretKey?: string) =>
             set((state) => {
-              state.account.publicKey = value;
+              const defaultSecretKey = secretKey || "";
+              state.account.publicKey = publicKey;
+              state.account.secretKey = defaultSecretKey;
+              state.account.registeredNetwork = state.network;
```

Comment:
We can simplify the secret key assignment inline.

```suggestion
              state.account.publicKey = publicKey;
              state.account.secretKey = secretKey || "";
              state.account.registeredNetwork = state.network;
```